src/index.js

```diff
@@ -57,35 +59,60 @@ class ExtractTextPlugin {
 
   mergeNonInitialChunks(chunk, intoChunk, checkedChunks) {
     if (!intoChunk) {
-      checkedChunks = [];
-      chunk.getChunks().forEach((c) => {
-        if (isInitialOrHasNoParents(c)) return;
-        this.mergeNonInitialChunks(c, chunk, checkedChunks);
-      }, this);
-    } else if (checkedChunks.indexOf(chunk) < 0) {
-      checkedChunks.push(chunk);
-      chunk.forEachModule((module) => {
-        intoChunk.addModule(module);
-        module.addChunk(intoChunk);
-      });
-      chunk.getChunks().forEach((c) => {
-        if (isInitialOrHasNoParents(c)) return;
-        this.mergeNonInitialChunks(c, intoChunk, checkedChunks);
-      }, this);
+      const newCheckedChunks = [];
+
+      for (const asyncChunk of chunk.getAllAsyncChunks()) {
+        if (!asyncChunk.isOnlyInitial()) {
+          this.mergeNonInitialChunks(asyncChunk, chunk, newCheckedChunks);
+        }
+      }
+    } else if (!checkedChunks.includes(chunk)) {
+      const newCheckedChunks = checkedChunks.concat(chunk);
+
+      for (const chunkModule of chunk.modulesIterable) {
+        intoChunk.addModule(chunkModule);
+        chunkModule.addChunk(intoChunk);
+      }
+
+      for (const asyncChunk of chunk.getAllAsyncChunks()) {
+        if (!asyncChunk.isOnlyInitial()) {
+          this.mergeNonInitialChunks(asyncChunk, intoChunk, newCheckedChunks);
+        }
+      }
     }
   }
 
-  renderExtractedChunk(chunk) {
+  static renderExtractedChunk(chunk) {
     const source = new ConcatSource();
-    chunk.forEachModule((module) => {
-      const moduleSource = module.source();
-      source.add(
-        this.applyAdditionalInformation(
-          moduleSource,
-          module.additionalInformation
-        )
-      );
-    }, this);
+
+    for (const chunkModule of chunk.modulesIterable) {
+      let moduleSource = chunkModule.source();
```

Comment:
Seems this code causes runtime error when chunkModule is `ConcatenatedModule` instance (https://github.com/webpack/webpack/blob/master/lib/optimize/ConcatenatedModule.js#L370). `ConcatenatedModule`'s `source` function requires dependencyTemplate and runtimeTemplate parameters.

Reply:
Can you please share the code that causes this runtime error when compiled? If the code isn’t public, could you please try removing as much code as possible so as to still exit with a runtime error?

Thanks!

Reply:
@log2-hwan Please open an issue about this with some logs if possible